cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -699,3 +699,150 @@ func TestNodeGroupMachineSetDeleteNodes(t *testing.T) {
 		t.Fatalf("expected 5 nodes, got %v", actual)
 	}
 }
+
+func TestNodeGroupMachineSetDeleteNodesWithMismatchedNodeGroups(t *testing.T) {
+	nodes := make([]*apiv1.Node, 2)
+	machines := make([]*v1beta1.Machine, 2)
+	nodeObjects := make([]runtime.Object, 2)
+	machineObjects := make([]runtime.Object, 2)
+
+	machineSet1 := &v1beta1.MachineSet{
+		TypeMeta: v1.TypeMeta{
+			Kind: "MachineSet",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "machineset1",
+			Namespace: "test-namespace",
+			UID:       "machineSet1",
+		},
+		Spec: v1beta1.MachineSetSpec{
+			Replicas: int32ptr(int32(len(machines) / 2)),
+		},
+	}
+
+	machineSet2 := &v1beta1.MachineSet{
+		TypeMeta: v1.TypeMeta{
+			Kind: "MachineSet",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "machineset2",
+			Namespace: "test-namespace",
+			UID:       "machineSet2",
+		},
+		Spec: v1beta1.MachineSetSpec{
+			Replicas: int32ptr(int32(len(machines) / 2)),
+		},
+	}
+
+	for i := 0; i < len(nodes); i++ {
+		refMachineSet := machineSet1
+		if i > 0 {
```

Comment:
Did you mean `i%2==0`?

Reply:
The len is only 2, so after the first iteration only.